### PR TITLE
Show an error dialog if the EFI partition is too small

### PR DESCRIPTION
**Pull request by mmstick**
If an EFI partition is too small

- `dialog-warning-symbolic` will be set on the partition
- a tooltip will be set on the partition to explain the issue
- the partition will not be added to the list of mounts

screenshot from 2018-05-31 15-20-46

### CODE REVIEW

**Comment by cassidyjames** [comment]
@mmstick instead of an error dialog, can we prevent users from making this mistake in the first place? My initial thoughts are:

- Badge the too-small EFI partition with a `dialog-warning-symbolic` icon
- Add a tooltip that says it must be 256 MB
- Mention that requirement in the bottom copy, where we list the other requirements
- Don't enable the "Erase & Install" button until that requirement is met

Generally leading users to prevent errors will result in a less frustrating UX than throwing an error dialog.

**Comment by mmstick** [comment]
@cassidyjames Sure, I can look into that.

**Comment by mmstick** [comment]
@cassidyjames It is done.

**Comment by cassidyjames** [comment]
Hm, I think we'll want to force that icon to a certain contrasting color, but I'm not sure how to do that off the top of my head. We should also probably do something about it getting cut off, though that's not unique to this branch…

File: src/Widgets/PartitionMenu.vala [hunk 1]
```diff
@@ -280,22 +280,37 @@ public class Installer.PartitionMenu : Gtk.Popover {
...
-            Gtk.IconSize.SMALL_TOOLBAR
-        );
+        var mount_icon_name = error == null
+            ? "process-completed-symbolic"
+            : "dialog-warning-symbolic";
+
+        var icon_theme = Gtk.IconTheme.get_default ();
+        var pixbuf = icon_theme.load_icon (mount_icon_name, 16, 0);
+
+        if (error != null) {
```
Comment (cassidyjames) [inline src/Widgets/PartitionMenu.vala]: It looks like we can just do this in the Gtk CSS theme which is probably the best and most flexible route. i.e. for the storage bars, we can set `-gtk-icon-palette: success #fff, warning #fff, error #fff;` to make sure it contrasts, or we can even set it to different colors for each filesystem type.

**Pull request merged**